app/models/ann_data_file_info.rb

```diff
@@ -106,9 +106,22 @@ def merge_form_fragments(form_data, fragments)
 
   # find a data_fragment of a given type based on arbitrary key/value pairs
   # any key/value pairs that don't match return false and fail the check for :detect
+  # also supports finding values as both strings and symbols (for data_type values)
   def find_fragment(**attrs)
     data_fragments.detect do |fragment|
-      !{ **attrs }.map { |k, v| fragment[k] == v }.include?(false)
+      !{ **attrs }.map { |k, v| fragment[k] == v || fragment[k] == v.send(transform_for(v)) }.include?(false)
```

Comment:
I had forgotten I sent this code to you... I included it on my draft PR that covers expression parsing (due to the same bug we found).  Depending on who merges first, we'll need to remove or deal with the conflict then.